andela/ah-backend-realers
# 166187849 Setup and deploy base application to Heroku

#### What does this PR do?
- Setup and deploy base application to Heroku

#### Description of Task to be completed?

- add Procfile file to define process type
- add runtime.txt to define python version
- install packages to run the app in production
- add configuration settings to support the app to run in production/staging on heroku

#### How should this be manually tested?
- Kindly visit click [here](https://dashboard.heroku.com/pipelines/96f9db0b-7b9e-4bed-a5b8-31a20b907fa1) to log into the heroku dashboard to view the review apps.
- Using the credentials I shared in our channels for username & password.
<img width="1293" alt="Screenshot 2019-06-05 at 17 07 15" src="https://user-images.githubusercontent.com/21230051/58962724-764afe00-87b4-11e9-8d24-8f1cbe0413f8.png">

- Click on the links below to confirm that the staging and production links are working.
- [staging](https://ah-backend-realers-staging.herokuapp.com/) url
- [production](https://ah-backend-realers.herokuapp.com/)  url

#### Any background context you want to provide?
The project currently had no heroku setup for continuous delivery and with no review apps for the raise PR. so with this PR the integration with set this in our project.

#### What are the relevant pivotal tracker stories?

- Story ID - [166187849](https://www.pivotaltracker.com/story/show/166187849)


#### DEV NOTE
Kindly note that I have changed the SECRET KEY  & DEBUG to pick value from the environment variable; therefore after the branch has been merged on develop, add the following below in you env file : -
- ```export IS_DEBUG_ON=True``` - can be ```True or False```
- ```export SECRET_KEY='anySecretKey'```
Also, remember to do `pip install -r requirements.txt` to install the Heroku dependency packages
